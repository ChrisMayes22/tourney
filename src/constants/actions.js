export const INITIALIZE_CHARACTER = 'INITIALIZE_CHARACTER';
export const REMOVE_CHARACTER = 'REMOVE_CHARACTER';
export const REMOVE_LOWEST_SCORE = 'REMOVE_LOWEST_SCORE';
export const RATE_CHARACTER = 'RATE_CHARACTER';
export const SUBMIT_CHARACTER_RATINGS = 'SUBMIT_CHARACTER_RATINGS';
export const SET_PLAYERS = 'SET_PLAYERS';
export const RESET_FOR_FINALS = 'RESET_FOR_FINALS';
export const CHOOSE_WINNER = 'CHOOSE_WINNER';

export const initializeCharacter = function(imageUrl, id){
    return {type: INITIALIZE_CHARACTER, 
        payload: {image: imageUrl, id}}
}
export const removeCharacter = function(id){
    console.log('Remove Character Dispatched')
    console.log(REMOVE_CHARACTER)
    return {type: REMOVE_CHARACTER, id}
}
export const removeLowestScore = function(){
    return {type: REMOVE_LOWEST_SCORE};
}
export const rateCharacter = function(points, rowId, character){
    return {type: RATE_CHARACTER, payload:{
        points, rowId, character}}
}
export const submitCharacterRatings = function(playersArray){
    return {type: SUBMIT_CHARACTER_RATINGS, payload:{
        players: playersArray
    }};
};
export const setPlayers = function(num){
    return {type: SET_PLAYERS, number: num}
};
export const resetForFinals = function(){
    return {type: RESET_FOR_FINALS}
}
export const chooseWinner = function(character){
    return {type: CHOOSE_WINNER, character}
}

