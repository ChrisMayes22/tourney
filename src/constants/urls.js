export const VOTING_PAGE = '/voting';
export const ELIMINATION_PAGE = '/elimination';
export const NOT_ENOUGH_CHARACTERS = '/not-enough-characters';
export const UPLOAD_PAGE = '/upload';
export const WINNER_PAGE = '/winner';
export const HOME = '/';