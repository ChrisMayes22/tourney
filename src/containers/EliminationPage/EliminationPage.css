.flexContainer__col{
    display: flex;
    margin: 5px auto;
    flex-direction: column;
    width: 95vw;
    height: 75vh;
    align-items: center;
    background-color: #a63939;
    background-color: #a63939;
    color: rgb(253, 211, 250);
    font-size: 1.3rem;
    border-radius: 10px;
}

h3{
    margin: 1.2rem 0 0.8rem 0;
}

h1, h5{
    margin: 0.3rem;
}

.grid {
    margin: 5px auto;
    display: grid;
    grid-template-rows: 15vh;
    grid-template-columns: 19vw 19vw 19vw 19vw 19vw;
    grid-template-areas: "header header header header header";
    grid-auto-rows: 30vh;
    grid-auto-columns: minmax(auto, 19vw);
    background-color: rgb(253, 211, 250);
    justify-items: center;
    align-items: center;
}

.mainImg{
    max-height: 50vh;
    max-width: 72vw;
    -webkit-filter: grayscale(100%); /* Safari 6.0 - 9.0 */
    filter: grayscale(100%);
}

.refImg{
    height: auto;
    max-height: 28vh;
    max-width: 18vw;
    border-radius: 15px;
    box-shadow: 5px 4px 3px rgba(0, 0, 0, 0.7);
    -webkit-filter: grayscale(100%); /* Safari 6.0 - 9.0 */
    filter: grayscale(100%);
}

.otherLosersHeader{
    grid-area: header;
    font-size: 2rem;
    color: #a63939;
}

.eliminationHeader{
    margin: 2rem;
}