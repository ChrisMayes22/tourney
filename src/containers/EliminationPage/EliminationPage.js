import React, { Component } from 'react';
import classes from './EliminationPage.css';
import SubmitButton from '../../components/SubmitButton/SubmitButton';
import {removeLowestScore } from '../../constants/actions';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import * as urls from '../../constants/urls';

export class EliminationPage extends Component{

    render(){
        return(
            <React.Fragment>
                <main className={classes.flexContainer__col}>
                    <h1 className={classes.eliminationHeader}>ELIMINATED!!</h1>
                    <div className={classes.mainImageContainer}>
                        {[this.props.characters.filter(char => char.isEliminated.check)
                            .reduce(function(accumulator, currentValue){
                                if(accumulator.isEliminated.whenEliminated < currentValue.isEliminated.whenEliminated) {
                                    return currentValue;
                                } else {
                                    return accumulator;
                                }
                            })].map(function(char){ 
                                return(<img src={char.imageUrl} id={char.id} key={`${char.id}-key`} alt="The most recent losing competitor"/>);
                            })}
                    </div>
                    <Link  to={this.props.characters.filter(char => !char.isEliminated.check).length < 7 ? urls.FINALS_PAGE : urls.ELIMINATION_PAGE}
                        onClick={this.props.characters.filter(char => !char.isEliminated.check).length > 6 ? this.props.onEliminate : null}>
                        <SubmitButton 
                            id={'EliminationPage__continueButton'}
                            children={this.props.characters.filter(char => !char.isEliminated.check).length < 7 ? 
                                'On to the finals!' : 'Eliminate another!'}/>
                    </Link>
                </main>
                <section className={classes.grid}>
                    <h5 className={classes.otherLosersHeader}>~ The Other Losers ~</h5>
                    {this.props.characters.filter(char => char.isEliminated.check).length > 1? this.props.characters.filter(char => char.isEliminated.check).filter(
                        char => char.id !== (this.props.characters.reduce(function(accumulator, currentValue){
                            if(accumulator.isEliminated.whenEliminated < currentValue.isEliminated.whenEliminated) {
                                return currentValue;
                            } else {
                                return accumulator;
                            }
                        })).id).map(function(char){
                                return (<img src={char.imageUrl} key={char.id} className={classes.refImg} alt="a losing competitor"/>)
                        }) : null}
                </section>
            </React.Fragment>
        );
    }
}

const mapDispatchToProps = dispatch => {
    return {
        onEliminate: () => dispatch(removeLowestScore())
    }
}

let mapStateToProps = state => {
    return {
        characters: state.characters
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(EliminationPage);