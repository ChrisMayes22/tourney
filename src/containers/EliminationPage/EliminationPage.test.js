import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter(), disableLifecycleMethods: true });

import React from 'react';
import { shallow } from 'enzyme';

import { EliminationPage } from './EliminationPage';
import * as urls from '../../constants/urls';

describe('When unconnected EliminationPage first renders', function(){
  let props = {characters: [{
    imageUrl: `foo$Image`,
    roundRatings: {
        rowOne: {rowId: `foo$id1-row-one`, points: 5},
        rowTwo: {rowId: `foo$id1-row-two`, points: 5},
        rowThree: {rowId: `foo$id1-row-three`, points: 5},
        rowFour: {rowId: `foo$id1-row-four`, points: 5}
    },
    points: 20,
    hadTurn: false,
    isEliminated: {check: false, whenEliminated: null},
    id: 'foo$id1'
  },
  {
    imageUrl: `foo$Image`,
    roundRatings: {
        rowOne: {rowId: `foo$id2-row-one`, points: 1},
        rowTwo: {rowId: `foo$id2-row-two`, points: 1},
        rowThree: {rowId: `foo$id2-row-three`, points: 1},
        rowFour: {rowId: `foo$id2-row-four`, points: 1}
    },
    points: 4,
    hadTurn: false,
    isEliminated: {check: true, whenEliminated: 0},
    id: 'foo$id2'
  },
  {
    imageUrl: `foo$Image`,
    roundRatings: {
        rowOne: {rowId: `foo$id3-row-one`, points: 2},
        rowTwo: {rowId: `foo$id3-row-two`, points: 2},
        rowThree: {rowId: `foo$id3-row-three`, points: 2},
        rowFour: {rowId: `foo$id3-row-four`, points: 2}
    },
    points: 8,
    hadTurn: false,
    isEliminated: {check: true, whenEliminated: 1},
    id: 'foo$id3'
  },
  {
    imageUrl: `foo$Image`,
    roundRatings: {
        rowOne: {rowId: `foo$id3-row-one`, points: 3},
        rowTwo: {rowId: `foo$id3-row-two`, points: 4},
        rowThree: {rowId: `foo$id3-row-three`, points: 3},
        rowFour: {rowId: `foo$id3-row-four`, points: 3}
    },
    points: 13,
    hadTurn: false,
    isEliminated: {check: true, whenEliminated: 2},
    id: 'foo$id4'
  }],
  players: 3}

    test('EliminationPage renders as expected', () => {
      const wrapper = shallow(<EliminationPage {...props}/>)
    
      expect(wrapper).toMatchSnapshot();
    })

    describe('Character maps render correctly', function(){
      let wrapper = shallow(<EliminationPage {...props}/>)
      test('Exactly one character renders inside the Main Image container', function(){
        expect( wrapper.find('[alt="The most recent losing competitor"]')).toHaveLength(1);
      })
      test('The character with the highest isEliminated.whenEliminated value renders in the Main Image container', function(){
        const mostRecentElim = wrapper.find('[alt="The most recent losing competitor"]');
        const id = mostRecentElim.prop('id');
        expect(id).toBe('foo$id4');
      })
      test('The number of characters in the `other losers` map === the number of eliminated characters -1', function(){
        expect(wrapper.find('[alt="a losing competitor"]')).toHaveLength(2);
      })
      test('Given that there is only one eliminated character, the `other losers map` should not render, but main container should', function(){
        props = {characters: [{
          imageUrl: `foo$Image`,
          roundRatings: {
              rowOne: {rowId: `foo$id1-row-one`, points: 1},
              rowTwo: {rowId: `foo$id1-row-two`, points: 1},
              rowThree: {rowId: `foo$id1-row-three`, points: 1},
              rowFour: {rowId: `foo$id1-row-four`, points: 1}
          },
          points: 4,
          hadTurn: false,
          isEliminated: {check: true, whenEliminated: 0},
          id: 'foo$id1'
        }]}

        wrapper = shallow(<EliminationPage {...props}/>)

        expect(wrapper.find('[alt="a losing competitor"]')).toHaveLength(0);
        expect(wrapper.find('[alt="The most recent losing competitor"]')).toHaveLength(1);
      })
    })
    describe('Link urls render as expected', function(){
      test('Given there are 7 or more non-eliminated characters, only link to urls.ELIMINATION_PAGE should exist', function(){
        props = {characters: [{isEliminated:{check: true}},{isEliminated:{check: false}},{isEliminated:{check: false}},
                  {isEliminated:{check: false}},{isEliminated:{check: false}},{isEliminated:{check: false}},
                  {isEliminated:{check: false}},{isEliminated:{check: false}}]};

        const wrapper = shallow(<EliminationPage {...props}/>)

        expect(wrapper.exists(`[to="${urls.ELIMINATION_PAGE}"]`)).toBe(true);
        expect(wrapper.exists(`[to="${urls.FINALS_PAGE}"]`)).toBe(false);
      })      
      test('Given there are 6 or fewer non-eliminated characters, only link to urls.FINALS_PAGE should exist', function(){
        props = {characters: [{isEliminated:{check: true}},{isEliminated:{check: false}},{isEliminated:{check: false}},
                  {isEliminated:{check: false}},{isEliminated:{check: false}},{isEliminated:{check: false}},
                  {isEliminated:{check: false}}]};

        const wrapper = shallow(<EliminationPage {...props}/>)

        expect(wrapper.exists(`[to="${urls.ELIMINATION_PAGE}"]`)).toBe(false);
        expect(wrapper.exists(`[to="${urls.FINALS_PAGE}"]`)).toBe(true);
      })  
    })
    describe('Conditionally rendered text renders as expected', function(){
      test('Given there are 6 or fewer non-eliminated characters, Continue Button should have text `On to the finals!`', function(){
        const wrapper = shallow(<EliminationPage {...props}/>);

        const text = wrapper.find('#EliminationPage__continueButton').prop('children');

        expect(text).toEqual('On to the finals!')
      })
      test('Given there are 7 or more non-eliminated characters, Continue Button should have text `Eliminate another!On to the finals!`', function(){
        props = {characters: [{isEliminated:{check: true}},{isEliminated:{check: false}},{isEliminated:{check: false}},
          {isEliminated:{check: false}},{isEliminated:{check: false}},{isEliminated:{check: false}},
          {isEliminated:{check: false}},{isEliminated:{check: false}}]};
        const wrapper = shallow(<EliminationPage {...props}/>);

        const text = wrapper.find('#EliminationPage__continueButton').prop('children');

        expect(text).toEqual('Eliminate another!')
      })
    })

  })