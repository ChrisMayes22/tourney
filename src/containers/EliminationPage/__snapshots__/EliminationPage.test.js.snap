// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When unconnected EliminationPage first renders EliminationPage renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EliminationPage
    characters={
      Array [
        Object {
          "hadTurn": false,
          "id": "foo$id1",
          "imageUrl": "foo$Image",
          "isEliminated": Object {
            "check": false,
            "whenEliminated": null,
          },
          "points": 20,
          "roundRatings": Object {
            "rowFour": Object {
              "points": 5,
              "rowId": "foo$id1-row-four",
            },
            "rowOne": Object {
              "points": 5,
              "rowId": "foo$id1-row-one",
            },
            "rowThree": Object {
              "points": 5,
              "rowId": "foo$id1-row-three",
            },
            "rowTwo": Object {
              "points": 5,
              "rowId": "foo$id1-row-two",
            },
          },
        },
        Object {
          "hadTurn": false,
          "id": "foo$id2",
          "imageUrl": "foo$Image",
          "isEliminated": Object {
            "check": true,
            "whenEliminated": 0,
          },
          "points": 4,
          "roundRatings": Object {
            "rowFour": Object {
              "points": 1,
              "rowId": "foo$id2-row-four",
            },
            "rowOne": Object {
              "points": 1,
              "rowId": "foo$id2-row-one",
            },
            "rowThree": Object {
              "points": 1,
              "rowId": "foo$id2-row-three",
            },
            "rowTwo": Object {
              "points": 1,
              "rowId": "foo$id2-row-two",
            },
          },
        },
        Object {
          "hadTurn": false,
          "id": "foo$id3",
          "imageUrl": "foo$Image",
          "isEliminated": Object {
            "check": true,
            "whenEliminated": 1,
          },
          "points": 8,
          "roundRatings": Object {
            "rowFour": Object {
              "points": 2,
              "rowId": "foo$id3-row-four",
            },
            "rowOne": Object {
              "points": 2,
              "rowId": "foo$id3-row-one",
            },
            "rowThree": Object {
              "points": 2,
              "rowId": "foo$id3-row-three",
            },
            "rowTwo": Object {
              "points": 2,
              "rowId": "foo$id3-row-two",
            },
          },
        },
        Object {
          "hadTurn": false,
          "id": "foo$id4",
          "imageUrl": "foo$Image",
          "isEliminated": Object {
            "check": true,
            "whenEliminated": 2,
          },
          "points": 13,
          "roundRatings": Object {
            "rowFour": Object {
              "points": 3,
              "rowId": "foo$id3-row-four",
            },
            "rowOne": Object {
              "points": 3,
              "rowId": "foo$id3-row-one",
            },
            "rowThree": Object {
              "points": 3,
              "rowId": "foo$id3-row-three",
            },
            "rowTwo": Object {
              "points": 4,
              "rowId": "foo$id3-row-two",
            },
          },
        },
      ]
    }
    players={3}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <main>
          <h1>
            ELIMINATED!!
          </h1>
          <div>
            <img
              alt="The most recent losing competitor"
              src="foo$Image"
            />
          </div>
          <Link
            onClick={null}
            replace={false}
            to="/finals"
          >
            <submitButton>
              On to the finals!
            </submitButton>
          </Link>
        </main>,
        <section>
          <h5>
            ~ The Other Losers ~
          </h5>
          <img
            alt="a losing competitor"
            src="foo$Image"
          />
          <img
            alt="a losing competitor"
            src="foo$Image"
          />
        </section>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              ELIMINATED!!
            </h1>,
            <div>
              <img
                alt="The most recent losing competitor"
                src="foo$Image"
              />
            </div>,
            <Link
              onClick={null}
              replace={false}
              to="/finals"
            >
              <submitButton>
                On to the finals!
              </submitButton>
            </Link>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ELIMINATED!!",
              "className": undefined,
            },
            "ref": null,
            "rendered": "ELIMINATED!!",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="The most recent losing competitor"
                  src="foo$Image"
                />,
              ],
              "className": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "foo$id4",
                "nodeType": "host",
                "props": Object {
                  "alt": "The most recent losing competitor",
                  "className": undefined,
                  "src": "foo$Image",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <submitButton>
                On to the finals!
              </submitButton>,
              "onClick": null,
              "replace": false,
              "to": "/finals",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "On to the finals!",
              },
              "ref": null,
              "rendered": "On to the finals!",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": "main",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h5>
              ~ The Other Losers ~
            </h5>,
            Array [
              <img
                alt="a losing competitor"
                src="foo$Image"
              />,
              <img
                alt="a losing competitor"
                src="foo$Image"
              />,
            ],
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "~ The Other Losers ~",
              "className": undefined,
            },
            "ref": null,
            "rendered": "~ The Other Losers ~",
            "type": "h5",
          },
          Object {
            "instance": null,
            "key": "foo$id2",
            "nodeType": "host",
            "props": Object {
              "alt": "a losing competitor",
              "className": undefined,
              "src": "foo$Image",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": "foo$id3",
            "nodeType": "host",
            "props": Object {
              "alt": "a losing competitor",
              "className": undefined,
              "src": "foo$Image",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "section",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <main>
            <h1>
              ELIMINATED!!
            </h1>
            <div>
              <img
                alt="The most recent losing competitor"
                src="foo$Image"
              />
            </div>
            <Link
              onClick={null}
              replace={false}
              to="/finals"
            >
              <submitButton>
                On to the finals!
              </submitButton>
            </Link>
          </main>,
          <section>
            <h5>
              ~ The Other Losers ~
            </h5>
            <img
              alt="a losing competitor"
              src="foo$Image"
            />
            <img
              alt="a losing competitor"
              src="foo$Image"
            />
          </section>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                ELIMINATED!!
              </h1>,
              <div>
                <img
                  alt="The most recent losing competitor"
                  src="foo$Image"
                />
              </div>,
              <Link
                onClick={null}
                replace={false}
                to="/finals"
              >
                <submitButton>
                  On to the finals!
                </submitButton>
              </Link>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ELIMINATED!!",
                "className": undefined,
              },
              "ref": null,
              "rendered": "ELIMINATED!!",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt="The most recent losing competitor"
                    src="foo$Image"
                  />,
                ],
                "className": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "foo$id4",
                  "nodeType": "host",
                  "props": Object {
                    "alt": "The most recent losing competitor",
                    "className": undefined,
                    "src": "foo$Image",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <submitButton>
                  On to the finals!
                </submitButton>,
                "onClick": null,
                "replace": false,
                "to": "/finals",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "On to the finals!",
                },
                "ref": null,
                "rendered": "On to the finals!",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": "main",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h5>
                ~ The Other Losers ~
              </h5>,
              Array [
                <img
                  alt="a losing competitor"
                  src="foo$Image"
                />,
                <img
                  alt="a losing competitor"
                  src="foo$Image"
                />,
              ],
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "~ The Other Losers ~",
                "className": undefined,
              },
              "ref": null,
              "rendered": "~ The Other Losers ~",
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": "foo$id2",
              "nodeType": "host",
              "props": Object {
                "alt": "a losing competitor",
                "className": undefined,
                "src": "foo$Image",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": "foo$id3",
              "nodeType": "host",
              "props": Object {
                "alt": "a losing competitor",
                "className": undefined,
                "src": "foo$Image",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "section",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
